src/OrchardCore/OrchardCore.ResourceManagement.Abstractions/ResourceDefinition.cs

```diff
@@ -334,8 +335,13 @@ public override bool Equals(object obj)
 
         public override int GetHashCode()
         {
-            return (Name ?? "").GetHashCode() ^ (Type ?? "").GetHashCode();
+            return HashCode.Combine(Name, Type);
```

Comment:
The new HashCode class should pretty much used always, https://github.com/dotnet/corefx/issues/14354